freeipa/client/init.sls

```diff
@@ -6,6 +6,77 @@ include:
 - freeipa.client.nsupdate
 - freeipa.client.cert
 
+{%- if pillar.freeipa.client.install_principal is defined %}
+{% set otp = salt['random.get_str'](20) %}
+
+push_principal:
```

Comment:
It's good to prefix all resources with formula name (eg. freeipa_client_principal_file) to avoid conflict with other states' resources